src/binding.cc

```diff
@@ -828,6 +828,14 @@ bool v8__ArrayBuffer__IsDetachable(const v8::ArrayBuffer& self) {
   return ptr_to_local(&self)->IsDetachable();
 }
 
+bool v8__ArrayBuffer__WasDetached(const v8::ArrayBuffer& self) {
+  /// v8 does not provide a direct way to check if an ArrayBuffer
+  /// is already detached. Instead, we use JSArrayBufferView->WasDetached
+  return v8::Utils::OpenHandle(
+             local_to_ptr(v8::Uint8Array::New(ptr_to_local(&self), 0, 0)))
+      ->WasDetached();
```

Comment:
```suggestion
  return v8::Utils::OpenHandle(&self)->was_detached();
```

Reply:
Great catch, just tested it, and it's faster as well.